docs/docs/addons-test-utils.md

```diff
@@ -0,0 +1,323 @@
+---
+id: test-utils
+title: Test Utilities
+layout: docs
+category: Reference
+permalink: docs/test-utils.html
+prev: animation.html
+next: create-fragment.html
+---
+
+**Importing**
+
+```javascript
+import ReactTestUtils from 'react-addons-test-utils' // ES6
+var ReactTestUtils = require('react-addons-test-utils') // ES5
+```
+
+## Overview
+
+`ReactTestUtils` makes it easy to test React components in the testing framework of your choice. At Facebook we use [Jest](https://facebook.github.io/jest/) for painless JavaScript testing. Learn how to get started with Jest through the Jest website's [React Tutorial](http://facebook.github.io/jest/docs/tutorial-react.html#content).
+
+> Note:
+>
+> Airbnb has released a testing utility called Enzyme, which makes it easy to assert, manipulate, and traverse your React Components' output. If you're deciding on a unit testing utility to use together with Jest, or any other test runner, it's worth checking out: [http://airbnb.io/enzyme/](http://airbnb.io/enzyme/)
+
+### Simulating Events
+
+`Simulate` is possibly the single most useful utility in `ReactTestUtils`. It allows you to simulate an event dispatch on a DOM node.
+
+ - [`Simulate`](#simulate)
+
+### Other APIs
+
+ - [`renderIntoDocument()`](#renderintodocument)
+ - [`mockComponent()`](#mockcomponent)
+ - [`isElement()`](#iselement)
+ - [`isElementOfType()`](#iselementoftype)
+ - [`isDOMComponent()`](#isdomcomponent)
+ - [`isCompositeComponent()`](#iscompositecomponent)
+ - [`isCompositeComponentWithType()`](#iscompositecomponentwithtype)
+ - [`findAllInRenderedTree()`](#findallinrenderedtree)
+ - [`scryRenderedDOMComponentsWithClass()`](#scryrendereddomcomponentswithclass)
+ - [`findRenderedDOMComponentWithClass()`](#findrendereddomcomponentwithclass)
+ - [`scryRenderedDOMComponentsWithTag()`](#scryrendereddomcomponentswithtag)
+ - [`findRenderedDOMComponentWithTag()`](#findrendereddomcomponentwithtag)
+ - [`scryRenderedComponentsWithType()`](#scryrenderedcomponentswithtype)
+ - [`findRenderedComponentWithType()`](#findrenderedcomponentwithtype)
+
+### Shallow Rendering (Experimental)
```

Comment:
It is used for testing in myriad of apps at this point and I would not call it experimental. Yes, it has some rough edges but it works fairly well. 

I would mention Enzyme here again and direct people to its shallow() doc. Enzyme provides nicer API for shallow rendering.